rpcserver.go

```diff
@@ -7218,6 +7220,57 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 			"for channel with ID %d: %w", channelID, err)
 	}
 
+	// If this is a hodl invoice, then we'll copy over the relevant fields,
+	// then route this through the invoicerpc instead.
+	if req.HodlInvoice != nil {
+		payHash, err := lntypes.MakeHash(req.HodlInvoice.PaymentHash)
+		if err != nil {
+			return nil, fmt.Errorf("error creating payment hash: %w", err)
+		}
+
+		peerPub, err := btcec.ParsePubKey(peerPubKey[:])
+		if err != nil {
+			return nil, fmt.Errorf("error parsing peer pubkey: %w", err)
+		}
+
+		hopHint := []zpay32.HopHint{
+			{
+				NodeID:      peerPub,
+				ChannelID:   acceptedQuote.Scid,
+				FeeBaseMSat: uint32(inboundPolicy.FeeBaseMsat),
+				FeeProportionalMillionths: uint32(
+					inboundPolicy.FeeRateMilliMsat,
+				),
+				CLTVExpiryDelta: uint16(inboundPolicy.TimeLockDelta),
+			},
+		}
+
+		payReq, err := r.cfg.Lnd.Invoices.AddHoldInvoice(
+			ctx, &invoicesrpc.AddInvoiceData{
+				Memo:            iReq.Memo,
+				Value:           lnwire.MilliSatoshi(iReq.ValueMsat),
+				Hash:            &payHash,
+				DescriptionHash: iReq.DescriptionHash,
+				Expiry:          iReq.Expiry,
+				Private:         true,
```

Comment:
Should we take `private` from `iReq` as well?

Reply:
not sure of the side effects, but since we're here we assume we're using the scid hophint flow?

is it even a valid move from user perspective to set `Private: false`?

Reply:
So revisiting things, I think we actually want to set `Private` to false: https://github.com/lightningnetwork/lnd/blob/4f6b510869abd03a2defabd57257b0bc94f4e557/lnrpc/invoicesrpc/addinvoice.go#L442-L460

Notice how if `Private` is true, then we'll try to continue adding hop hints up the max limit. However, in this case, we only care about our singular hop hint, as we want to make sure the incoming HTLC goes over the specified channel. 

Pushed a fix up to set this to false.